File: storage/data_transpiled/at_coder/abc119/D/4905147.java
@ VisibleForTesting static String solve ( String string ) {
  int a = Integer . parseInt ( string ) ;
  int b = Integer . parseInt ( string ) ;
  int q = Integer . parseInt ( string ) ;
  int ... stx = Integer . parseInt ( string ) ;
  int bignum = 3 * 10 * 10 ;
  int [ ] s = new int [ stx . length ] ;
  for ( int i = 0 ;
  i < stx . length ;
  i ++ ) {
    s [ i ] = - bignum ;
  }
  int [ ] t = new int [ stx . length ] ;
  for ( int i = a ;
  i < stx . length ;
  i ++ ) {
    t [ i ] = - bignum ;
  }
  for ( int i = a ;
  i < stx . length ;
  i ++ ) {
    int _x = stx [ i - q ] ;
    int iS = Integer . parseInt ( s [ i ] ) ;
    int iT = Integer . parseInt ( t [ i ] ) ;
    int ls = _x - s [ iS - 1 ] ;
    int rs = s [ iS ] - _x ;
    int lt = _x - t [ iT - 1 ] ;
    int rt = t [ iT ] - _x ;
    ans [ i ] = Integer . toString ( Math . min ( Math . max ( rs , rt ) , Math . max ( ls , lt ) ) + rt + ls + Math . min ( rt , ls ) + rs + lt + Math . min ( rs , lt ) ) ;
  }
  return String . join ( "\n" , ans ) ;
}

File: storage/data_original/at_coder/abc119/D/4905147.py
from bisect import bisect_left


def solve(string):
    a, b, q, *stx = map(int, string.split())
    bignum = 3 * 10**10
    s = [-bignum] + stx[:a] + [bignum]
    t = [-bignum] + stx[a:a + b] + [bignum]
    ans = []
    for _x in stx[-q:]:
        i_s, i_t = bisect_left(s, _x), bisect_left(t, _x)
        ls, rs = _x - s[i_s - 1], s[i_s] - _x
        lt, rt = _x - t[i_t - 1], t[i_t] - _x
        ans.append(str(min(max(rs, rt), max(ls, lt), rt + ls + min(rt, ls),
                           rs + lt + min(rs, lt))))
    return "\n".join(ans)


if __name__ == '__main__':
    n, m, l = map(int, input().split())
    print(solve('{} {} {}\n'.format(n, m, l) + '\n'.join([input() for _ in range(n + m + l)])))